### PR TITLE
Update logic of separated by comma fields, refactoring of tests

### PR DESCRIPTION
**Pull request by voscob**
### Description

I updated the logic of the comma-separated fields and refactored the tests.
I've added new unit tests for the configuration fields.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-clickhouse/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config/source.go [hunk 1]
```diff
@@ -58,37 +58,59 @@ func ParseSource(cfg map[string]string) (Source, error) {
 		BatchSize:      defaultBatchSize,
 	}
 
-	keyColumns := strings.Split(cfg[KeyColumns], ",")
-	for i := range keyColumns {
-		if keyColumn := strings.TrimSpace(keyColumns[i]); keyColumn != "" {
+	if cfg[KeyColumns] != "" {
+		keyColumns := strings.Split(cfg[KeyColumns], ",")
```
Comment (maksenius) [inline config/source.go:62]: Maybe we can do TrimSpace here. What do you think?
`keyColumns := strings.Split(TrimSpace(cfg[KeyColumns]), ",")`

File: config/source.go [hunk 2]
```diff
@@ -58,37 +58,59 @@ func ParseSource(cfg map[string]string) (Source, error) {
...
 	if err != nil {
-		return Source{}, fmt.Errorf("validate source config: %w", err)
+		return Source{}, err
+	}
+
+	if len(sourceConfig.Columns) == 0 {
+		return sourceConfig, nil
+	}
+
+	columnsMap := make(map[string]struct{}, len(sourceConfig.Columns))
```
Comment (maksenius) [inline config/source.go:101]: maybe we can remove this variable and check key column and ordering column inside loop?

File: config/source.go [hunk 3]
```diff
@@ -58,37 +58,59 @@ func ParseSource(cfg map[string]string) (Source, error) {
...
+			if keyColumn == "" {
+				return Source{}, fmt.Errorf("invalid %q", KeyColumns)
+			}
+
 			sourceConfig.KeyColumns = append(sourceConfig.KeyColumns, keyColumn)
 		}
 	}
 
 	if cfg[Columns] != "" {
 		columnsSl := strings.Split(cfg[Columns], ",")
```
Comment (maksenius) [inline config/source.go:74]: Maybe we can do TrimSpace here. What do you think?
сolumns := strings.Split(TrimSpace(cfg[Columns]), ",")

File: config/source.go (hunk 1 above)
Comment (voscob) [reply config/source.go:62]: We can, but I don't know why, because we'll be trimming the spaces in the next range.

File: config/source.go (hunk 2 above)
Comment (voscob) [reply config/source.go:101]: We can check that `columns` contains the `orderingColumn` in a range of `columns`.
But I must check that the `columns` contains all `keyColumns` keys.

File: config/source.go (hunk 1 above)
Comment (voscob) [reply config/source.go:62]: But we have to trim each key after the split.

**Review by maksenius — APPROVED** [review]

File: config/source.go (hunk 1 above)
Comment (voscob) [reply config/source.go]: Done.

**Pull request merged**